demos/tests/run_tests.py

```diff
@@ -206,17 +218,70 @@ def option_to_args(key, value):
                         continue
 
                     for device, dev_arg in device_args.items():
-                        print('Test case #{}/{}:'.format(test_case_index, device),
+                        print('\nTest case #{}/{}:'.format(test_case_index, device),
                             ' '.join(shlex.quote(str(arg)) for arg in dev_arg + case_args))
                         print(flush=True)
                         try:
                             start_time = timeit.default_timer()
                             subprocess.check_output(fixed_args + dev_arg + case_args,
                                 stderr=subprocess.STDOUT, universal_newlines=True)
                             execution_time = timeit.default_timer() - start_time
-                        except subprocess.CalledProcessError as e:
-                            print(e.output)
-                            print('Exit code:', e.returncode)
+
+                            # 'if' is dev-only, won't be needed when all demos will be compatible with ssim test
+                            if '-o' in case_args and case_args[case_args.index('-o') + 1] != '.':
+                                similarity_res = []
+
+                                out_folder = case_args[case_args.index('-o') + 1]
+                                demo_out_file = str(Path(out_folder) / 'out_%08d.BMP')
+                                demo_raw_file = str(Path(out_folder) / 'raw_%08d.BMP')
+                                out_cap = cv.VideoCapture(demo_out_file)
+                                raw_cap = cv.VideoCapture(demo_raw_file)
+                                if not out_cap.isOpened() or not raw_cap.isOpened():
+                                    raise RuntimeError("Unable to open input files.")
+                            
+                                while True:
+                                    out_ret, out_frame = out_cap.read()
+                                    raw_ret, raw_frame = raw_cap.read()
+
+                                    if out_ret and raw_ret:
+                                        crop = CROP_SIZE[demo.full_name]
+                                        height, width = out_frame.shape[:2]
+                                        out_frame = out_frame[crop[0] : height - crop[2], crop[3] : width - crop[1]]
+                                        raw_frame = raw_frame[crop[0] : height - crop[2], crop[3] : width - crop[1]]
+                                        similarity = list(map(lambda x: round(x, 3),
```

Comment:
I think we shouldn't store such long numbers as `0.9666283495334497` in `reference_values.json`, because the method used for comparison is not so precise and thresholds are also about `10^-3`. It might confuse someone. So, I round to `0.967`.